### ISSUE TITLE
Display Cards are not accepted on Windows AVS device

### ISSUE THREAD

**Issue by jc78** [alexa/alexa-avs-sample-app#1098]
I've been using this sample app on Windows 10 with no issues. But, I just recently went into my device's setup and added the **Display Cards with Media** capability hoping to be able to receive and process the RenderTemplate directives sent to my device.

My plan is to test and develop my device on Win10 because it's easier to iterate on then on a Raspberry Pi. I'll eventually move my work to a Raspi later on in my development process.

Unfortunately, after enabling the feature, I'm receiving the message (both spoken to me by Alexa & on the sample app UI) that _"The target device does not support directives for the Display interface"_. This only happens when I send a RenderTemplate directive from my skill. Just to be sure, I was looking at the code (in the sample app) that receives directives and verified that I am seeing a TemplateRuntime:RenderTemplate directive. But I'm not sure if this is the same directive I'm sending from my skill, or a generic one because it contains the same message as above in the payload, and none of the metadata I sent from my skill.

However, I can ask Alexa _"Who is Usain Bolt"_ and will get a RenderTemplate returned without the error message spoken to me or in the payload. So Display Cards appear to be working properly when the directive is delivered from Alexa, but not from my skill.

In the Amazon Developer Console, under my skill's information tab, I have ensured that the Render Template option is turned onto "yes". I've tested the intent in the Service Simulator and verified that the Service Request payload contains the proper device info stating that my device supports displays:
```
      "device": {
        "supportedInterfaces": {
          "Display": {
            "templateVersion": "1",
            "markupVersion": "1"
          }
        }
      }
```
And, this is the format of my JSON that is returned from skill. It contains the RenderTemplate JSON, that I assume is in the proper format.
```
{
  "version": "1.0",
  "response": {
    "outputSpeech": {
      "text": "OK",
      "type": "PlainText"
    },
    "card": {
      "content": "SessionSpeechlet - OK",
      "title": "SessionSpeechlet - ShowRenderTemplate"
    },
    "reprompt": {
      "outputSpeech": {
        "text": "OK",
        "type": "PlainText"
      }
    },
    "speechletResponse": {
      "outputSpeech": {
        "text": "OK"
      },
      "card": {
        "content": "SessionSpeechlet - OK",
        "title": "SessionSpeechlet - ShowRenderTemplate"
      },
      "directives": [
        {
          "template": {
            "title": "My Template Title",
            "textContent": {
              "primaryText": {
                "richText": "Template Body Text"
              }
            },
            "token": "MyTemplateTestToken",
            "backButtonBehavior": "HIDDEN"
          }
        }
      ],
      "reprompt": {
        "outputSpeech": {
          "text": "OK"
        }
      },
      "shouldEndSession": true
    }
  },
  "sessionAttributes": {}
}
```

Am I missing a step in the process? Are Display Cards supported on the Win10 version of the sample app? Or do I need to use the Raspi version? Could this be a compatibility problem with the Win10 Sample AVS App combined with a custom skill?

I've searched for information and tutorials on how to use this, but could only find some high-level info on how to enable the feature in the developer console. Nothing on what you need to do within the sample app code to make it work.

Any help would be appreciated.

**Comment by jc78** [comment]
I have confirmed that my same skill properly displays my RenderTemplate on an Echo Show. At this point, I'm running on the assumption of a comparability issue with a Win10 AVS device and a custom skill.

I'll try to move my AVS project over to my Raspberry Pi and try again with that combination. Hopefully that will work.

**Comment by jc78** [comment]
It appears as if running on a Raspi also fails to accept the RenderTemplate directives from my skill, but will accept them from Alexa.

I'm assuming that this is a compatibility issue with AVS devices and third party skills.

I would love for somebody to confirm this. Even better, point me to a solution. :-)

**Comment by dhpp** [comment]
Hi @jc78,

As a first step, have you enabled this in the Developer Console for your device?  If you click 'Manage' on your particular device, then look under 'Capabilities', there is a 'Display Cards' option.  If this is not enabled, then please do so, and allow maybe 1 hour for propagation of the setting and try again.  Let us know if this solves your issue, thanks.

**Comment by jc78** [comment]
Yes, that is enabled. I enabled it Wednesday evening and tried to make use of the feature Thursday morning.
![image](https://user-images.githubusercontent.com/27864662/35163793-c887b09e-fcfc-11e7-9f5d-5f44eec75ced.png)

Like I mentioned, the device is properly receiving cards from Alexa 1st party requests. It fails to receive cards from my custom skill or any other third party skill that delivers cards. I verified this with another skill from the skills store.

I tried again this morning (Friday morn) and its still an issue.

**Comment by bandkenamazoncom** [comment]
Thanks @jc78,

Can you attach the logs from samples\javaclient\logs\application.log, where you show one from your skill and one from a first party Alexa skill?
